open-widget-framework/src/widget-form.js

```diff
@@ -33,22 +33,25 @@ class WidgetForm extends Component {
     /**
      * Fetch available widget classes and configurations after component mount
      */
-    this.props.widgetFrameworkSettings.fetchData(this.props.fetchRoute)
-      .then(this.resetForm)
-      .catch(this.props.widgetFrameworkSettings.errorHandler)
+     loadData()
   }
 
   componentDidUpdate(prevProps) {
     /**
      * Fetch new form data if the fetchRoute changes
      */
     if (prevProps.fetchRoute !== this.props.fetchRoute) {
-      this.props.widgetFrameworkSettings.fetchData(this.props.fetchRoute)
-        .then(this.resetForm)
-        .catch(this.props.widgetFrameworkSettings.errorHandler)
+      loadData()
```

Comment:
should this be a call to `this.loadData()`?

Reply:
if so we might have to bind `this.loadData` in the constructor

Reply:
I believe it should, and I did.